frontend/lib/js/user/selectors.js

```diff
@@ -4,23 +4,25 @@ export function selectPermissions(state) {
     : null;
 }
 
-export function canDownloadResult(state) {
+export function canDownloadResult(state, datasetId?: string) {
   const permissions = selectPermissions(state);
 
   if (!permissions) return false;
 
-  const permission = permissions.find(
-    p => p.domains.includes("datasets") && p.abilities.includes("download")
-  );
-
   const { selectedDatasetId } = state.datasets;
 
-  return (
-    !!permission &&
-    (permission.targets.includes("*") ||
-      // TODO: This makes not that much sense yet, because
-      //       the dataset shouldn't be the selected one, but the one
-      //       which relates to the result
-      permission.targets.includes(selectedDatasetId))
+  // TODO: Fallback to selectedDatasetId probably doesn't make that much sense,
+  //       because the dataset to use should probably always relates to the result
+  //       or entity, that is being checked here.
+  //
+  const dataset = datasetId || selectedDatasetId;
+
+  const permission = permissions.find(
```

Comment:
Zur vollen Unterstützung der Wildcardpermissions müsste das denke ich so aus sehen:
```
p =>
      (p.domains.includes("datasets") ||  p.domains.includes("*")) &&
      (p.abilities.includes("download") ||  p.abilities.includes("*") || (p.abilities.isEmpty() && !p.domains.isEmpty())) &&
      (p.targets.includes(dataset) || p.targets.includes("*") || (p.targets.isEmpty() && !p.abilities.isEmpty()))
```

Reply:
Ja, ginge, aber dann wäre das hier in der spezifischen `canDownload`-Funktion halb-allgemein gelöst und müsste bei allen weiteren Abfragen kopiert werden => Ich würde dafür gerne eine Abstraktion bauen und es gleich richtig implementieren 

=> Dauert noch etwas. Wenn ich richtig verstehe ist das zumindest noch nicht zeitkritisch, richtig?

Reply:
Side-Notiz ;-) 
<img width="417" alt="Bildschirmfoto 2020-01-20 um 21 31 27" src="https://user-images.githubusercontent.com/1403093/72756002-40844980-3bcc-11ea-87cc-3696c55bb3ce.png">

Reply:
Sorry war Pseudo-Code :)

Reply:
> => Dauert noch etwas. Wenn ich richtig verstehe ist das zumindest noch nicht zeitkritisch, richtig?

Nicht umbedingt, aber wir würden gerne die Woche Live gehen. Denke spätestens Donnerstag weil Freitag live gehen immer doof ist. gglg

Reply:
Ah, bis Donnerstag schaffe ich das nicht, frühestens am Samstag falls das mit Wildcards Priorität vor all den anderen Dingen hat. Der PR hier ist von mir aus merge-bar.